service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java

```diff
@@ -56,7 +65,7 @@
                     .getValue()
                     .stream()
                     .mapToInt(molecularProfile ->
-                        molecularProfileSampleCountSet.getOrDefault(molecularProfile.getStableId(), 0))
+                        BooleanUtils.isTrue(molecularProfile.getPatientLevel()) ? molecularProfilePatientCountSet.getOrDefault(molecularProfile.getStableId(), 0) : molecularProfileSampleCountSet.getOrDefault(molecularProfile.getStableId(), 0))
```

Comment:
Just a thought - rather than having this `isTrue` boilerplate everywhere, why not add that logic and an `@NotNull` annotation to the getter in MolecularProfile. The current approach seems like a bug waiting to happen, especially when you consider line 908 of StudyViewFilterApplier.

Reply:
This is a good idea and it's true, I will add this annotation!